draft-ietf-tls-tls13.md

```diff
@@ -1567,8 +1567,8 @@ cipher_suites
   the server does not recognize, support, or wish to use, the server
   MUST ignore those cipher suites, and process the remaining ones as
   usual. Values are defined in {{cipher-suites}}. If the client is
-  attempting a PSK key establishment, it MUST only advertise the single
-  cipher suite associated with the PSK.
+  attempting a PSK key establishment, it MUST advertise at least one
+  cipher suite containing a Hash associated with the PSK.
```

Comment:
Weren't you objecting in your thread that asymmetric MUSTs are confusing? It seems this particular asymmetric MUST doesn't serve much purpose and only risks encouraging servers towards an unsound decision algorithm.

Hash vs ciphers was an intentional change to align the rules for external PSKs (where expecting an externally-configured AEAD is a bit much) and ticket-based PSKs. And, yes, it means 0-RTT has stricter requirements than resumption. That is already the case, and necessarily so, because 0-RTT cannot change ALPN or basically any property at all. One reasonable selection algorithm is:

1. Decide if you were going to resume or not.
2. Select all other parameters. ALPN, etc.
3. If all of the following are true, acknowledge 0-RTT. Otherwise, decline it.
   - We are resuming.
   - The client offered 0-RTT and we are willing to accept it.
   - The PSK is okay for use with 0-RTT.
   - All connection parameters associated with the PSK match those selected organically.